### ISSUE TITLE
fix_precision Does not Support CUDA

### ISSUE THREAD

**Issue by Psychofun** [OpenMined/PySyft#2401]
**Describe the bug**

 Convert  tensor allocated on GPU to fix precision tensor causes  error:
```
TypeError: can't convert CUDA tensor to numpy
```
 
**To Reproduce**
```
# To use syft with cuda.
th.set_default_tensor_type(th.cuda.FloatTensor) 
import syft as sy 
hook = sy.TorchHook(th)

weight = torch.tensor(2,32,64) # On gpu.
hidden = weight.new(2,32,64).zero_().fix_precision()
```
Then error:

```
TypeError: can't convert CUDA tensor to numpy. Use Tensor.cpu() to copy the tensor to host memory first.
```

**Expected behavior**
fix_precision() must to check if tensor is on GPU. Then use Tensor.detach().cpu().numpy()

Bug seems to be on:

. . \privateai\lib\site-packages\syft\frameworks\torch\tensors\interpreters\native.py", line 672, in _requires_large_precision

```  
    np.log2(np.abs(self.clone().detach().numpy()) + 1) + base_fractional > max_precision

``` 

**Screenshots**
*Error in console:*
![cmd](https://user-images.githubusercontent.com/36560929/61999189-60094200-b081-11e9-9be3-5298feff54f3.png)
*Model code:*
![code](https://user-images.githubusercontent.com/36560929/61999192-65668c80-b081-11e9-82fa-ef5e630dd71b.png)

**Desktop:**
 - OS:  Windows
 - Version: 10

**Additional context**

**Comment by wentaiwu92** [comment]
Seems that PySyft does not support GPU(cuda) yet?

**Comment by Psychofun** [comment]
> 
> 
> Seems that PySyft does not support GPU(cuda) yet?

In general it does. Only when uses zero_().fix_precision() for encryption.

**Comment by wentaiwu92** [comment]
> > Seems that PySyft does not support GPU(cuda) yet?
> 
> In general it does. Only when uses zero_().fix_precision() for encryption.

However, I tried to send my model to cuda using:
```
model.to(torch.device('cuda:0'))
```
and I got an error that goes: 
**RuntimeError: Expected object of backend CPU but got backend CUDA for argument #2 ‘source’**

But the above only happens when you hook _PySyft_ with _Pytorch_ - I tried to run that line of code without the hook code `hook = sy.TorchHook(th)`  and no error prompted.

So I suppose this error is caused by an validation check in _PySyft_ which expects the model's parameters to be on CPU, rather than GPU(cuda).

**Further**
I didn't find any evidence in the codes under PySyft/examples/tutorials that utilize Cuda or invoke xxx.to(device) to copy tensors/models to the GPU. I love PySyft but I feel like cuda features are probably not supported yet. 

Hope the developers of PySyft could provide more details.
